### ISSUE TITLE
nixos defaultUserShell

### ISSUE THREAD

**Issue by dezren39** [oilshell/oil#2017]
hi i want to use ysh or maybe osh as my default shell in in nix, for nushell I just set it like so

```
defaultUserShell = pkgs.nushell;
```

doing the same doesn't work for oils.

I'm not sure if i need to make a nixos module, or maybe it's because oils-for-unix and oil both have multiple binaries? it may be that a package for ysh and osh is needed, or maybe the oils-for-unix package could have embedded functions? I dunno. I think I could hack this to work as a custom shell, but it would be nice if the nix package itself did it 'right'.

I know this isn't the nixpkgs repo, but it's uh.. a bit busy there. If I get time and find a solution I'll comment here.

**Comment by dezren39** [comment]
ps there is an oil and an oils-for-unix package, each unique. maybe keep the oil one to not affect existing configurations but oil could be an alias to oils-for-unix or something.

**Comment by dezren39** [comment]
ahh oil is the python compile? ok that answers one question

**Comment by dezren39** [comment]
maybe it would be good to have a home-manager or nixos module or both to configure default shopt and which pkgs between ysh/osh etc

**Comment by andychu** [comment]
Yes the difference between the two tarballs explained in the "packaging guidelines" - https://www.oilshell.org/release/0.22.0/

(the `oil` tarball will go away eventually)


I'm not familiar with home-manager but there are a couple dirs to make for history and rc files - https://www.oilshell.org/release/0.22.0/doc/getting-started.html

**Comment by dezren39** [comment]
works now dunno why
